cpp/include/cudf/table/row_operators.cuh

```diff
@@ -91,6 +91,26 @@ __device__ weak_ordering relational_compare(Element lhs, Element rhs)
   return detail::compare_elements(lhs, rhs);
 }
 
+/**
+ * @brief Compare the nulls according to null order.
+ *
+ * @param lhs_is_null boolean representing if lhs is null
+ * @param rhs_is_null boolean representing if lhs is null
+ * @param null_precedence null order
+ * @return weak_ordering Indicates the relationship between null in lhs and rhs columns.
+ */
+inline __device__ auto null_compare(bool lhs_is_null, bool rhs_is_null, null_order null_precedence)
+{
+  if (lhs_is_null and rhs_is_null) {  // null <? null
+    return weak_ordering::EQUIVALENT;
+  } else if (lhs_is_null) {  // null <? x
+    return (null_precedence == null_order::BEFORE) ? weak_ordering::LESS : weak_ordering::GREATER;
+  } else if (rhs_is_null) {  // x <? null
+    return (null_precedence == null_order::AFTER) ? weak_ordering::LESS : weak_ordering::GREATER;
```

Comment:
```suggestion
  } else if (lhs_is_null || rhs_is_null) {  // null <? x or x <? null
    return null_precedence == null_order::BEFORE ? weak_ordering::LESS : weak_ordering::GREATER;
```

Reply:
For rhs_is_null, condition is `null_precedence == null_order::AFTER`

Reply:
oh right, nvm then.